google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java

```diff
@@ -247,6 +250,46 @@ public void readRowsAsync(Query query, ResponseObserver<Row> observer) {
     return stub.createReadRowsCallable(rowAdapter);
   }
 
+  /**
+   * Convenience method to asynchronously mutate a single row atomically. Cells already present in
+   * the row are left unchanged unless explicitly changed by the {@link RowMutation}.
+   *
+   * <p>Sample code:
+   *
+   * <pre>{@code
+   * InstanceName instanceName = InstanceName.of("[PROJECT]", "[INSTANCE]");
+   * try (BigtableClient bigtableClient = BigtableClient.create(instanceName)) {
+   *   RowMutation mutation = RowMutation.create("[TABLE]", "[ROW KEY]")
+   *     .setCell("[FAMILY NAME]", "[QUALIFIER]", "[VALUE]");
+   *
+   *   ApitFuture<Void> future = bigtableClient.mutateRow(mutation);
+   * }
+   * }</pre>
+   */
+  public ApiFuture<Void> mutateRow(RowMutation rowMutation) {
```

Comment:
It seems like the name should contain `async` since it returns a future instead of the result.

In the pure autogen clients we generally don't provide methods like this; generally we provide the callable variant + synchronous variants.

Reply:
How strongly do you feel about this? My thought is that our customers will most likely want to use the async api, so I wanted to optimize for that usecase. The penalty to go from async to sync is less than going from callable to async:

```
// Proposed style:
client.mutateRow() // async
client.mutateRow().get() // sync

suggested style 1:
client.mutateRowAsync() // async
client.mutateRowAsync().get() // sync

// suggested style 2:
client.mutateRow() // sync
client.mutateRowCallable().futureCall() // async
```

I think that the overall style is close enough to not confuse customers

Reply:
Well, all of the other generated GAPIC clients are set up with sync being the basic case. Another option is this:

```
client.mutateRow() // sync
client.mutateRowAsync() // async

// alternate routes:
client.mutateRowAsync().get() // sync
client.mutateRowCallable().call() // sync
client.mutateRowCallable().futureCall() // async
```

Reply:
This will require changing existing methods, so I would like to address this rename in a separate PR I created an issue to track it:

https://github.com/GoogleCloudPlatform/google-cloud-java/issues/2917